pkgs/development/python-modules/clingo/default.nix

```diff
@@ -0,0 +1,42 @@
+{
+  lib,
+  buildPythonPackage,
+  python,
+  cffi,
+  clingo,
+  setuptools,
+}:
+
+buildPythonPackage rec {
+  inherit (clingo) pname version src;
+
+  postPatch = ''
+    cd libpyclingo
+    python compile.py h
+  '';
+
+  pyproject = true;
+  build-system = [ setuptools ];
+
+  dependencies = [ cffi ];
+  nativeBuildInputs = [ python ];
+  buildInputs = [ clingo ];
+
+  postInstall = ''
+    mkdir -p "$out/${python.sitePackages}/clingo/lib/"
+    cp -Tr "${clingo}/lib" "$out/${python.sitePackages}/clingo/lib/"
```

Comment:
Creating a symlink would be cheaper, though I wonder why we need to copy these at all. The wheel distributed under the `clingo` name looks like this:

```
clingo/
clingo/__init__.py
clingo/__main__.py
clingo/_clingo.cpython-312-x86_64-linux-gnu.so
clingo/_internal.py
clingo/application.py
clingo/ast.py
clingo/backend.py
clingo/clingo.h
clingo/clingo.hh
clingo/configuration.py
clingo/control.py
clingo/core.py
clingo/propagator.py
clingo/py.typed
clingo/script.py
clingo/solving.py
clingo/statistics.py
clingo/symbol.py
clingo/symbolic_atoms.py
clingo/theory.py
clingo/theory_atoms.py
clingo/util.py
clingo-5.7.1.dist-info/
clingo-5.7.1.dist-info/LICENSE.md
clingo-5.7.1.dist-info/METADATA
clingo-5.7.1.dist-info/RECORD
clingo-5.7.1.dist-info/WHEEL
clingo-5.7.1.dist-info/top_level.txt
clingo.libs/
```

Reply:
That bit of code was copied over from brainflow, which was necessary there because of FFI. I'll try removing it and seeing if it still works.

Reply:
Yup, that was unneccesary. Removing in next force-push.

Reply:
Except that the wheel metadata now says

```
lib/python3.12/site-packages/clingo_cffi_system-5.5.0.post6.dist-info/
lib/python3.12/site-packages/clingo_cffi_system-5.5.0.post6.dist-info/LICENSE.md
lib/python3.12/site-packages/clingo_cffi_system-5.5.0.post6.dist-info/METADATA
lib/python3.12/site-packages/clingo_cffi_system-5.5.0.post6.dist-info/RECORD
lib/python3.12/site-packages/clingo_cffi_system-5.5.0.post6.dist-info/WHEEL
lib/python3.12/site-packages/clingo_cffi_system-5.5.0.post6.dist-info/top_level.txt
```

Reply:
Yes. What's the issue with that?

Reply:
If anything pins it as a dependency it won't be recognized, due to the name mismatch.

Reply:
Okay, I'm not quite sure I get the issue. I'll probably just patch the pyproject.toml